src/filter/time.rs

```diff
@@ -11,32 +11,30 @@ pub enum TimeFilter {
 
 impl TimeFilter {
     fn from_str(ref_time: &SystemTime, s: &str) -> Option<SystemTime> {
-        humantime::parse_duration(s)
-            .map(|duration| *ref_time - duration)
+        s.parse::<Span>()
+            .and_then(|duration| {
+                Zoned::try_from(*ref_time).and_then(|zdt| zdt.checked_sub(duration))
+            })
             .ok()
             .or_else(|| {
-                DateTime::parse_from_rfc3339(s)
-                    .map(|dt| dt.into())
+                let local_tz = TimeZone::system();
+                s.parse::<Timestamp>()
+                    .map(|ts| ts.to_zoned(TimeZone::UTC))
                     .ok()
                     .or_else(|| {
-                        NaiveDate::parse_from_str(s, "%F")
-                            .ok()?
-                            .and_hms_opt(0, 0, 0)?
-                            .and_local_timezone(Local)
-                            .latest()
-                    })
-                    .or_else(|| {
-                        NaiveDateTime::parse_from_str(s, "%F %T")
-                            .ok()?
-                            .and_local_timezone(Local)
-                            .latest()
+                        s.parse::<DateTime>()
+                            .map(|dt| local_tz.to_ambiguous_zoned(dt))
+                            .and_then(|zdt| zdt.later())
+                            .ok()
                     })
                     .or_else(|| {
                         let timestamp_secs = s.strip_prefix('@')?.parse().ok()?;
-                        DateTime::from_timestamp(timestamp_secs, 0).map(Into::into)
+                        Timestamp::from_second(timestamp_secs)
+                            .map(|ts| ts.to_zoned(TimeZone::UTC))
```

Comment:
It's a little unfortunate that we go from a timestmp to a Zoned, and then  back to a SystemTime, which is basically the same as a timestamp. 

But I don't think converting everything else to Timestamp first would be a lot better.

Reply:
Yeah I think this is fine. I think it might read a little clearer/simpler by just unifying to `Timestamp`, but behaviorally, this is all the same.